waku/node/v2/wakunode2.nim

```diff
@@ -223,7 +223,7 @@ proc start*(node: WakuNode, conf: WakuNodeConf) {.async.} =
 ## Public API
 ##
 
-method init*(conf: WakuNodeConf): Future[WakuNode] {.async.} =
+method init*(T: type WakuNode, conf: WakuNodeConf): Future[T] {.async.} =
```

Comment:
Assuming you ran tests etc and that this doesn't collide with the other init fn (diff arity so probably not)

Reply:
It works on tests.